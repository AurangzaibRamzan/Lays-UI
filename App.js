import React, { Component } from 'react';
import styled from 'styled-components/native';

const Container = styled.View`
  flex: 1;
  background-color: #F1B11B;
  justify-content: center;
  align-items: center;
`;

const Title = styled.Text`
  font-size: 20px;
  color: #fff;
  font-family:'HelveticaNeueBd';
`;
export default class App extends Component {
  render() {
    return (
      <Container >
        <Title>{'Welcome to React Native Lays UI!'}</Title>
      </Container>
    );
  }
}
