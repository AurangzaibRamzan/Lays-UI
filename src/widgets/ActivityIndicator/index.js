import React, { Component } from 'react';
import PropTypes from 'prop-types';

import {
  ActivityInicatorView,
} from './styles';

export default class ActivityIndicatorNative extends Component {
  render() {
    const { ...restProps } = this.props;
    return (
      <ActivityInicatorView  {...restProps} />
    )
  }
}

ActivityIndicatorNative.defaultProps = {
  color: "#0000ff",
  size: 'small',
  backgroundColor: 'transparent',
};

ActivityIndicatorNative.propsTypes = {
  color: PropTypes.string,
  size: PropTypes.oneOfType([
    PropTypes.string,
    PropTypes.number,
  ]),
  backgroundColor: PropTypes.string,
};