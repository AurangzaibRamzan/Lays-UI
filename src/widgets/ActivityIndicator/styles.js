import styled from 'styled-components/native';

export const ActivityInicatorView = styled.ActivityIndicator`
  background-color: ${props => props.backgroundColor || 'transparent'}
`;