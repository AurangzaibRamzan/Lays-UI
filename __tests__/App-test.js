import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';

import App from '../App';

// widget testing 

import ActivityIndicator from '../src/widgets/ActivityIndicator';

it('renders correctly', () => {
  renderer.create(<App />);
});

it('ActivityIndicator', () => {
  const tree = renderer.create(<ActivityIndicator />);
  expect(tree).toMatchSnapshot();
});