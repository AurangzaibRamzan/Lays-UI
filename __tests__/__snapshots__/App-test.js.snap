// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ActivityIndicator 1`] = `
<ActivityIndicator
  animating={true}
  backgroundColor="transparent"
  color="#0000ff"
  hidesWhenStopped={true}
  size="small"
  style={
    Array [
      Object {
        "backgroundColor": "transparent",
      },
    ]
  }
/>
`;
